lib/sensu/client/process.rb

```diff
@@ -465,17 +466,19 @@ def setup_json_socket
         options = @settings[:client][:socket] || Hash.new
         options[:bind] ||= "127.0.0.1"
         options[:port] ||= 3030
-        @logger.debug("binding client tcp and udp sockets", :options => options)
-        @sockets << EM::start_server(options[:bind], options[:port], Socket) do |socket|
-          socket.logger = @logger
-          socket.settings = @settings
-          socket.transport = @transport
-        end
-        @sockets << EM::open_datagram_socket(options[:bind], options[:port], Socket) do |socket|
-          socket.logger = @logger
-          socket.settings = @settings
-          socket.transport = @transport
-          socket.protocol = :udp
+        unless options[:enabled] == false
```

Comment:
few things, rather than checking a negative value and then negating it, I generally find it easier to read as checking for the positive value instead. Also I think that disabling this should be logged so I'd write it something like this:
```ruby
if options[:enabled]
  # insert existing logic
else
  # log that client socket is being disabled per configuration.
end
```

Reply:
Another option would be to do something like this:
```ruby
if !options[:enabled]
  # log socket disabled
else
  # insert existing logic
end
```

Reply:
The bottom one forces negation to ensure that `nil` values are evaluated as true (current default)

Reply:
The following illustrates the difference between the two:
```ruby
$ irb
irb(main):001:0> options = {}
=> {}
irb(main):002:0> if options[:foo] then p 'true' else p 'false' end
"false"
=> "false"
irb(main):003:0> if !options[:foo] then p 'true' else p 'false' end
"true"
=> "true"
irb(main):004:0> if options[:foo] || options[:foo].nil? then p 'true' else p 'false' end
"true"
=> "true"
```

You could also include an `|| options[:enabled].nil?` to catch this

Reply:
`if options[:foo] || options[:foo].nil?` is the most verbose and imo happens to be much easier to ready and understand.

Reply:
Hey @majormoses  and @portertech, thanks for the quick review! I pushed a change based on your feedback. My Ruby skills are very basic so, let me know if anything looks even slightly wrong. 

Should I also update `setup_http_socket` to use a similar logic/logging setup?

Reply:
@portertech your call but I think it's worth the minimal effort to ensure that they act the same way. We could also split them up as 2 separate PRs if you'd feel better about that.